File: Java/src/FourSumTest.java
import org.junit.Assert;
import org.junit.Test;

import java.util.Arrays;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class FourSumTest {

    @Test
    public void testFourSum() {
        FourSum fs = new FourSum();

        List<List<Integer>> list0 = fs.fourSum(new int[]{1, 0, -1, 0, -2, 2}, 0);
        Set<List<Integer>> set = new HashSet<>();
        set.add(Arrays.asList(-1, 0, 0, 1));
        set.add(Arrays.asList(-2, -1, 1, 2));
        set.add(Arrays.asList(-2, 0, 0, 2));
        Assert.assertTrue(list0.size() == 3);
        Assert.assertEquals(new HashSet<>(list0), set);

        set.clear();
        List<List<Integer>> list1 = fs.fourSum(new int[]{-5, 5, 4, -3, 0, 0, 4, -2}, 4);
        set.add(Arrays.asList(-5, 0, 4, 5));
        set.add(Arrays.asList(-3, -2, 4, 5));
        Assert.assertEquals(new HashSet<>(list1), set);
    }

}

File: Kotlin/src/FourSumTest.kt
import org.junit.Test

class FourSumTest {

    @Test
    fun testFourSum() {
        val fs = FourSum()

        val list0 = fs.fourSum(intArrayOf(1, 0, -1, 0, -2, 2), 0)
        assert(list0.size == 3)
        assert(list0.containsAll(listOf(listOf(-1, 0, 0, 1), listOf(-2, -1, 1, 2), listOf(-2, 0, 0, 2))))

        val list1 = fs.fourSum(intArrayOf(-5, 5, 4, -3, 0, 0, 4, -2), 4)
        assert(list1.size == 2)
        assert(list1.containsAll(listOf(listOf(-5, 0, 4, 5), listOf(-3, -2, 4, 5))))
    }

}